packages/react/src/wallet/adapters/ConcreteWallet.ts

```diff
@@ -155,6 +158,52 @@ export class ConcreteWallet extends Wallet {
     }
   }
 
+  async signFrameAction<TData extends TypedDataShape>(
```

Comment:
I can't figure out the type here for `_signTypedData` later to work with `data.domain`, `data.types` etc. below, so I introduced TypedDataShape, let me know what would be a better way, to make TypedData the implementation detail here.

```
signer._signTypedData(
        unsignedAction.data.domain,
        unsignedAction.data.types,
        unsignedAction.data.message,
      );
```